app/assets/src/components/visualizations/table/BaseTable.jsx

```diff
@@ -73,6 +74,10 @@ class BaseTable extends React.Component {
   handleColumnChange = selectedColumns => {
     const { protectedColumns } = this.props;
     this.setState({ activeColumns: concat(protectedColumns, selectedColumns) });
+    logAnalyticsEvent("BaseTable_column-selector_changed", {
```

Comment:
I worry that if we use BaseTable across the app, it can be difficult to distinguish the analytics event of a particular BaseTable. Not sure what we can do about that...Maybe have `BaseTable` take a `logId` prop that it logs? So I can set `logId="SamplesTable"` when I use it in the samples page (bad example) and the BaseTable will pass that id to snowflake.

Reply:
Same concern goes for all generic components like this.

Reply:
I share the concern. That's why in general I tried to log in the outermost component, wrapping passed-in callbacks. In this case, the behavior is entirely internal to `BaseTable`. So, as a fallback, we need to look at the url path that is logged with every analytics event to establish context.

That said, the argument could be made in this case we really do care more about the column-select behavior across pages.

Reply:
Ah, the URL is captured as well. That should help.